### PR TITLE
HttpLoggingInterceptor: improve detection of binary request/response bodies

### PR DESCRIPTION
**Pull request by DavidJRobertson**
See #5412 

Now checking the content type to see if it looks like a binary type when deciding whether or not to omit the body content from the logs.

### CODE REVIEW

File: okhttp-logging-interceptor/src/main/java/okhttp3/logging/HttpLoggingInterceptor.kt [hunk 1]
```diff
@@ -285,4 +288,23 @@ class HttpLoggingInterceptor @JvmOverloads constructor(
     return !contentEncoding.equals("identity", ignoreCase = true) &&
         !contentEncoding.equals("gzip", ignoreCase = true)
   }
+    private fun mediaTypeIsProbablyBinary(mediaType: MediaType?): Boolean {
+        val type = mediaType?.type
+        val subType = mediaType?.subtype
+
+        val binaryTypes = listOf("audio", "binary", "font", "image", "video")
+        val binaryApplicationSubtypes = listOf("binary", "excel", "java-archive", "java-vm", "lzx",
```
Comment (swankjesse) [inline okhttp-logging-interceptor/src/main/java/okhttp3/logging/HttpLoggingInterceptor.kt]: I very much do not want to maintain this list of binary media types. Wanna make this the user’s problem by using the same strategy as `headersToRedact`?

I think this also wants us to finally finish this: https://github.com/square/okhttp/issues/782

**Comment by yschimke** [comment]
@DavidJRobertson The other option here is for someone interested in this functionality to make a better HttpLoggingInterceptor as a library.  There are an infinite number of small features people want and having an active user/maintainer of this could help to point to a small focused library.  @JakeWharton has a ton of these small libraries and the model works well there. Maybe 2 small libraries, one with common media types, and the logger.

**Comment by yschimke** [comment]
The reason I specifically suggest the library approach here, is this sort of feature only really works if it's implemented as sensible defaults.  By the point we ask users to define setup calls to headersToRedact, they are likely to consider other options like implementing their own or using a library.  At the same time, I agree that we don't want the core library to be making arbitrary decisions like this.  Maybe the right thing to do here is actually to show hex output in a nice format?

**Comment by swankjesse** [comment]
Agreed. Closing for now.

**Pull request closed without merging**